### DISCUSSION TITLE
Candlestick chart date formatting

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#4641] [Help]
Hi,
I am trying to display a graph using the candlestick chart with a date in the x-axis. However I am not able to get it to display the date in proper format - it's displaying the date in JS value.
I have also tried to have the date value in formatted as yyyy-mm-dd string, but it's always converted to the JS value.

<img width="534" alt="Screenshot 2022-02-22 at 19 45 12" src="https://user-images.githubusercontent.com/7136209/155201201-17d4f85c-4e7d-4966-8c44-831492826199.png">

Is there a way to get around this?

**Comment by mjashanks** [comment]
Hey @micebrain 

It depends on what your datasource is. If you're using Budibase or SQL tables, then you could add a formula column - which returns your date value already formatted, as a string.

Because it's a string, you will need to format it like `YYYY-MM-DD`, so that it displays in order along the x axis.

So, with handlebars - `{{ date YourDateColumn "YYYY-MM-DD" }}`

If you are using a custom query or REST query, you should do this with a transformer. It's a little trickier to get this date formate in javascript, but I'm sure StackOverflow will tell you how!

**Reply by micebrain** [reply]
Thanks @mjashanks for this.
Actually I tested with transformer as I am using the API, I have my code as following:

```
const rows = data;
return rows.map((item) => {
	let new_item = Object.assign(item);
	let dt = new Date(Date.parse(item.date))
	new_item.date = dt.toISOString().split('T')[0];
	return new_item;
});
```

It works if I use line chart, unfortunately for candlestick chart the behaviour is different.
